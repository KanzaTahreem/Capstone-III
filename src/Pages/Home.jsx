import React from 'react';
import millify from 'millify';
import { useSelector } from 'react-redux';
// import Header from '../components/Header';
import headerImg from '../assets/images/header.png';
import GlobalData from '../components/GlobalData';
import '../styles/Header.css';
import '../styles/Main.css';

const Home = () => {
  const { globalArray } = useSelector((state) => state.global);
  return (
    <section className="homepage_container">
      <header>
        <div className="inner_header">
          <div>
            <h2>Hi, there ðŸ‘‹</h2>
            <p>
              Are you excited to kickstart your day with the fascinating world of cryptocurrencies?
              {' '}
              Let&apos;s dive into the fascinating world of cryptocurrencies together!
            </p>
          </div>
          <img src={headerImg} alt="girl working on laptop" />
        </div>
      </header>
      <main>
        <div className="global-data">
          <GlobalData
            activeCurrencies={millify(globalArray?.data?.active_cryptocurrencies)}
            endIcos={millify(globalArray?.data?.ended_icos)}
            onIcos={millify(globalArray?.data?.ongoing_icos)}
            comingIcos={millify(globalArray?.data?.upcoming_icos)}
            markets={millify(globalArray?.data?.markets)}
            marketCapPercent={millify(globalArray?.data?.market_cap_percentage.btc)}
            totalMarketCap={millify(globalArray?.data?.total_market_cap?.usd)}
            totalVolume={millify(globalArray?.data?.total_volume.btc)}
          />
        </div>
      </main>
    </section>
  );
};

export default Home;
