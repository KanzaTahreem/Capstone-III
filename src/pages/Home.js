import React from 'react';
import millify from 'millify';
import { useSelector } from 'react-redux';
import headerImg from '../assets/images/header.png';
import GlobalData from '../components/GlobalData';
import Loader from '../components/Loader';

const Home = () => {
  const { globalArray, loading, error } = useSelector((state) => state.global);

  if (loading) {
    return <Loader />;
  }

  if (error) {
    return <p>{error}</p>;
  }

  return (
    <section className="homepage_container">
      <header>
        <div className="inner_header">
          <div>
            <h2>Hi, there ðŸ‘‹</h2>
            <p>
              Are you excited to kickstart your day with the fascinating world of cryptocurrencies?
              {' '}
              Let&apos;s dive into the fascinating world of cryptocurrencies together!
            </p>
          </div>
          <img src={headerImg} alt="girl working on laptop" />
        </div>
      </header>
      <main>
        <div className="global-data">
          <GlobalData
            activeCurrencies={millify(globalArray?.data?.active_cryptocurrencies)}
            endIcos={millify(globalArray?.data?.ended_icos)}
            onIcos={millify(globalArray?.data?.ongoing_icos)}
            comingIcos={millify(globalArray?.data?.upcoming_icos)}
            markets={millify(globalArray?.data?.markets)}
            marketCapPercent={millify(globalArray?.data?.market_cap_percentage.btc)}
            totalMarketCap={millify(globalArray?.data?.total_market_cap?.usd)}
            totalVolume={millify(globalArray?.data?.total_volume.btc)}
          />
        </div>
      </main>
    </section>
  );
};

export default Home;
