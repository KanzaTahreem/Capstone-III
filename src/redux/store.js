import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './AllCoins/coins';
import chartReducer from './CoinChart/chart';

const store = configureStore({
  reducer: {
    crypto: coinsReducer,
    chart: chartReducer,
  },
});

export default store;
