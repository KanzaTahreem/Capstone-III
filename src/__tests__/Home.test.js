import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../pages/Home';

describe('Testing Home', () => {
  let home;
  beforeAll(async () => {
    home = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(home).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(home).toMatchSnapshot();
  });

  it('renders the component', () => {
    waitFor(() => expect(screen.getByText('Hi, there')).toBeInTheDocument());
    waitFor(() => expect(screen.getByText('Are you excited to kickstart your day')).toBeInTheDocument());
    waitFor(() => expect(screen.getAllByTestId('global-data-card')).toHaveLength(6));
  });
});
