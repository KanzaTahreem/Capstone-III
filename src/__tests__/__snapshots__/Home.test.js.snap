// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home should match the snapshot 1`] = `
<section
  className="homepage_container"
>
  <header>
    <div
      className="inner_header"
    >
      <div>
        <h2>
          Hi, there ðŸ‘‹
        </h2>
        <p>
          Are you excited to kickstart your day with the fascinating world of cryptocurrencies?
           
          Let's dive into the fascinating world of cryptocurrencies together!
        </p>
      </div>
      <img
        alt="girl working on laptop"
        src="header.png"
      />
    </div>
  </header>
  <main>
    <div
      className="global-data"
    >
      <div
        className="global-data-card"
      >
        <p>
          Active Currencies
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Ending Icos
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Ongoing Icos
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Upcoming Icos
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Total Markets
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          24h Market Change
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Total Market Cap
        </p>
        <p>
          undefined
        </p>
      </div>
      <div
        className="global-data-card"
        data-testId="global-data-card"
      >
        <p>
          Total Volume
        </p>
        <p>
          undefined
        </p>
      </div>
    </div>
  </main>
</section>
`;
